doc/ARS/Local_ARS_HLD.md

```diff
@@ -0,0 +1,892 @@
+
+```
+
+##### ACL_RULE changes
+
+```
+    container sonic-acl {
+
+        container ACL_RULE {
+
+            description "ACL_RULE part of config_db.json";
+
+            list ACL_RULE_LIST {
+
+                key "ACL_TABLE_NAME RULE_NAME";
+
+                leaf ACL_TABLE_NAME {
+                    type leafref {
+                        path "/acl:sonic-acl/acl:ACL_TABLE/acl:ACL_TABLE_LIST/acl:ACL_TABLE_NAME";
+                    }
+                }
+
+                leaf RULE_NAME {
+                    type string {
+                        length 1..255;
+                    }
+                } 
+...
+                leaf DISABLE_ARS_FORWARDING {
+                    description "Disable ARS forwarding on matching packets";
+                    type boolean;
+                    default false;
+                }
+            }
+         }
+    }
+```
+
+#### Config DB Enhancements  
+
+```
+; New container ARS_PROFILE_TABLE
+; ARS global configuration
+
+key                     = ARS_PROFILE|profile_name
+
+;field                  = value
+
+algorithm               = "ewma"        ;Path quality calculation algorithm
+max_flows               = uint32        ;Maximum number of flows that can be maintained for ARS
+sampling_interval       = uint32        ;Sampling interval in microseconds
+past_load_min_value     = uint16        ;Minimum value of Past load range.
+past_load_max_value     = uint16        ;Maximum value of Past load range.
+past_load_weight        = uint16        ;Weight of the past load
+future_load_min_value   = uint16        ;Minimum value of Future load range.
+future_load_max_value   = uint16        ;Maximum value of Future load range.
+future_load_weight      = uint16        ;Weight of the future load
+current_load_min_value  = uint16        ;Minimum value of Current load range.
+current_load_max_value  = uint16        ;Maximum value of Current load range.
+ipv4_enable             = boolean       ;Whether ARS is enabled over IPv4 packets
+ipv6_enable             = boolean       ;Whether ARS is enabled over IPv6 packets
+
+
+Configuration example:
+
+"ARS_PROFILE": {
+    "default": {
+        "algorithm": "ewma",
+        "max_flows" : "512",
+        "sampling_interval": "10",
+        "past_load_min_value" : "0",
+        "past_load_max_value" : "100",
+        "past_load_weight": "1",
+        "future_load_min_value" : "0",
+        "future_load_max_value" : "1000",
+        "future_load_weight": "5",
+        "ipv4_enable" : "true",
+        "ipv6_enable" : "true"
+    }
+}
+```
+
+```
+; New table ARS_INTERFACE_TABLE
+; ARS interfaces configuration
+
+key                      = ARS_INTERFACE|if_name          ;ifname is the name of the ARS-enabled interface
+
+;field                  = value
+
+scaling_factor          = uint32                          ;Port speed normalization
+
+Configuration example:
+
+"ARS_INTERFACE": {
+    "Ethernet0" : {
+        "scaling_factor": "100"
+    },
+    "Ethernet8" : {}

```

Comment:
without parameter will be used with default value for "scaling factor" or without it (if not used by other vendor's algo) ?